### PR TITLE
build a range of repo updates. Close: #35

### PR DESCRIPTION
**Pull request by M0nsteR**

### CODE REVIEW

File: src/knd_class.json.c [hunk 1]
```diff
@@ -179,9 +179,9 @@ static int export_conc_elem_JSON(void *obj,
 
     c->depth = 0;
     c->max_depth = 0;
-    if (self->max_depth) {
+    /*if (self->max_depth) {
```
Comment (NekoNe) [inline src/knd_class.json.c:182]: May be it should be deleted?
Why c->max_depth would be changed in export at all?

File: src/knd_class.json.c [hunk 2]
```diff
@@ -179,9 +179,9 @@ static int export_conc_elem_JSON(void *obj,
 
     c->depth = 0;
```
Comment (NekoNe) [inline src/knd_class.json.c:180]: Why do we change class state in the export function?

File: src/knd_repo.c [hunk 3]
```diff
@@ -472,38 +526,52 @@ static gsl_err_t present_repo_state(void *obj,
 
     task->type = KND_SELECT_STATE;
 
-    //if (!repo->states)                                     goto show_curr_state;
-
-    //latest_state = self->states;
-
-    //if (task->state_gt >= latest_state->numid)             goto show_curr_state;
+    if (!repo->updates)                                     goto show_curr_state;
```
Comment (NekoNe) [inline src/knd_repo.c]: Why do you use this strange formatting?

Comment (M0nsteR) [reply src/knd_repo.c]: I was having doubts about it myself, shall fix it.

File: src/knd_class.json.c (hunk 2 above)
Comment (M0nsteR) [reply src/knd_class.json.c:180]: The depth and max_depth fields are the relics of a non-concurrent implementation of a kndClass. They are used as temporary variables to reflect the need for a further expansion of class attributes while traversal and serialization. The plan is to replace these fields with recurrent, thread-safe variables.

File: src/knd_class.json.c (hunk 1 above)
Comment (M0nsteR) [reply src/knd_class.json.c:182]: The caller might wish to export objects with a desired depth of expansion, eg. expand the children of each class or grandchildren etc. The issue will be solved after implementing the concurrently accessed fields in kndClass.

**Pull request merged**